pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7497,179 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
 	result, _ := strconv.ParseBool(getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DECISIONS_ONLY"))
 	assert.False(t, result)
 }
+
+func TestCredentialWithCredentialSecretAndCredentialsUseAsDefault(t *testing.T) {
+	// with all credentials provided the behaviour use the legacy with username and password
```

Comment:
I'm not sure if I understand name of this test name correct. Should be credentials like admin name and password provided by user or use default values?

Reply:
TestCredentialsWithAllCredentialsSet (secretname, username, password)